### ISSUE TITLE
App crashes when selecting "star" in menu

### ISSUE THREAD

**Issue by mds08011** [commons-app/apps-android-commons#4742]
**Summary:** 

Every time I navigate to the "star" icon on the bottom menu, the device crashes.  I do not remember what the star icon represents.  Please see the video I created below.

**Steps to reproduce:** 

1. Open App
2. On the bottom menu with nearby places, etc, and tap the star icon
3. App crashes and asks to send logs.

**System logs:**

Logs are posted a separate comment below.

**Device and Android version:** 

Android version: 12
Device manufacturer: Google
Device model: Pixel 4a
User name: Mds08011
 
**Commons app version:** 

App version name: 3.1.1

**Screen-shots:** 

Video of how to reproduce the issue:

https://user-images.githubusercontent.com/8010709/144611252-30958a68-97bb-43a4-b9f5-e42618d4c6f7.mp4

**Would you like to work on the issue?**

No.

**Comment by mds08011** [comment]
```
USER_COMMENT=Tried to open the globe tab/icon
APP_VERSION_CODE=1025
APP_VERSION_NAME=3.1.1
ANDROID_VERSION=12
PHONE_MODEL=Pixel 4a
STACK_TRACE=android.database.sqlite.SQLiteException: no such column: location_exists (code 1 SQLITE_ERROR): , while compiling: SELECT location_name, location_language, location_description, location_category, location_label_text, location_label_icon, location_lat, location_long, location_image_url, location_wikipedia_link, location_wikidata_link, location_commons_link, location_pic, location_exists FROM bookmarksLocations
at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:1047)
at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:654)
at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:590)
at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:62)
at android.database.sqlite.SQLiteQuery.<init>(SQLiteQuery.java:37)
at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:46)
at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1546)
at android.database.sqlite.SQLiteQueryBuilder.query(SQLiteQueryBuilder.java:592)
at android.database.sqlite.SQLiteQueryBuilder.query(SQLiteQueryBuilder.java:463)
at fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsContentProvider.query(BookmarkLocationsContentProvider.java:61)
at android.content.ContentProvider.query(ContentProvider.java:1408)
at android.content.ContentProvider.query(ContentProvider.java:1504)
at android.content.ContentProvider$Transport.query(ContentProvider.java:272)
at android.content.ContentProviderClient.query(ContentProviderClient.java:196)
at android.content.ContentProviderClient.query(ContentProviderClient.java:178)
at android.content.ContentProviderClient.query(ContentProviderClient.java:168)
at fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsDao.getAllBookmarksLocations(BookmarkLocationsDao.java:48)
at fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsController.loadFavoritesLocations(BookmarkLocationsController.java:24)
at fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsFragment.initList(BookmarkLocationsFragment.java:85)
at fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsFragment.onResume(BookmarkLocationsFragment.java:78)
at androidx.fragment.app.Fragment.performResume(Fragment.java:2747)
at androidx.fragment.app.FragmentStateManager.resume(FragmentStateManager.java:363)
at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1197)
at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1354)
at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1432)
at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1495)
at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2617)
at androidx.fragment.app.FragmentManager.dispatchResume(FragmentManager.java:2581)
at androidx.fragment.app.Fragment.performResume(Fragment.java:2756)
at androidx.fragment.app.FragmentStateManager.resume(FragmentStateManager.java:363)
at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1197)
at androidx.fragment.app.FragmentManager.performPendingDeferredStart(FragmentManager.java:1109)
at androidx.fragment.app.FragmentManager.startPendingDeferredFragments(FragmentManager.java:1517)
at androidx.fragment.app.FragmentManager.doPendingDeferredStart(FragmentManager.java:2285)
at androidx.fragment.app.FragmentManager.execSingleAction(FragmentManager.java:1823)
at androidx.fragment.app.BackStackRecord.commitNowAllowingStateLoss(BackStackRecord.java:303)
at androidx.fragment.app.FragmentPagerAdapter.finishUpdate(FragmentPagerAdapter.java:238)
at androidx.viewpager.widget.ViewPager.populate(ViewPager.java:1244)
at androidx.viewpager.widget.ViewPager.populate(ViewPager.java:1092)
at androidx.viewpager.widget.ViewPager.onMeasure(ViewPager.java:1622)
at android.view.View.measure(View.java:25773)
at android.widget.RelativeLayout.measureChildHorizontal(RelativeLayout.java:735)
at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:481)
at android.view.View.measure(View.java:25773)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6980)
at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
at android.view.View.measure(View.java:25773)
at android.widget.RelativeLayout.measureChildHorizontal(RelativeLayout.java:735)
at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:481)
at android.view.View.measure(View.java:25773)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6980)
at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1552)
at android.widget.LinearLayout.measureVertical(LinearLayout.java:842)
at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
at android.view.View.measure(View.java:25773)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6980)
at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
at androidx.appcompat.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:143)
at android.view.View.measure(View.java:25773)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6980)
at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1552)
at android.widget.LinearLayout.measureVertical(LinearLayout.java:842)
at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
at android.view.View.measure(View.java:25773)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6980)
at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
at android.view.View.measure(View.java:25773)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6980)
at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1552)
at android.widget.LinearLayout.measureVertical(LinearLayout.java:842)
at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
at android.view.View.measure(View.java:25773)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6980)
at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
at com.android.internal.policy.DecorView.onMeasure(DecorView.java:761)
at android.view.View.measure(View.java:25773)
at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:3608)
at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:2407)
at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2677)
at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2126)
at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:8649)
at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1037)
at android.view.Choreographer.doCallbacks(Choreographer.java:845)
at android.view.Choreographer.doFrame(Choreographer.java:780)
at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1022)
at android.os.Handler.handleCallback(Handler.java:938)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loopOnce(Looper.java:201)
at android.os.Looper.loop(Looper.java:288)
at android.app.ActivityThread.main(ActivityThread.java:7838)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)

IS_SILENT=false
REPORT_ID=879446fc-63bf-4d81-ac9a-63d7269ad42e
USER_CRASH_DATE=2021-12-02T16:35:20.919-08:00
USER_EMAIL= 

```

**Comment by nicolas-raoul** [comment]
Thank you for the feedback!
I believe you installed from Play Store and only updated with Play Store, right?
While this kind of error is expected for people who (like me) switch between different versions, it should never happen to people who just follow Play Store (or F-Droid) updates.

**Comment by mds08011** [comment]
That is correct but after thinking about it more, I have a theory about what has been causing my problems. While my initial download and subsequent updates were all through the Play Store, I also changed phones in that time. I downloaded the app on my old phone and then used a USB cable to transfer data to my new phone (Google Pixel 2 to Pixel 4a). Since transferring phones, I had successfully used the Commons app to upload photos but I did not attempt to use the Nearby or Favorites features until recently when I filed this issue.  I think the data transfer caused me to run into the same problems one would expect from switching between different versions, as you mentioned.

The storage and cache clearing I did to resolve my problems with the "Nearby" feature (#4726) also resolved the app crashing I documented here. The crashing no longer occurs for me and you are welcome to close this issue if deemed a fairly obscure edge-case.
